### DISCUSSION TITLE
Custom Abadon suits

### DISCUSSION THREAD

**Discussion by kurotatsu77** [IHateMyKite/UnforgivingDevices#103] [Q&A] [answered]
Can we have example of esp/script in the doc folder of how to use this feature?
I've got a bit lost trying to figure out how to make my own patch suits...

**Comment by IHateMyKite** [comment]
I think example might be more powerful then words. So here is Patch I was working on for Latex Witch. It doesn't contain meshes, so the patch is not working (they are big, and I have no permission to share them yet). Just check the .esp and script.
[UnforgivingDevices - Latex Witch Patch.zip](https://github.com/IHateMyKite/UnforgivingDevices/files/10148465/UnforgivingDevices.-.Latex.Witch.Patch.zip)

**Comment by IHateMyKite** [comment]
Basically the main gist is that you create new quest, add the new script which extends UD_PatchInit and fill it correctly. In script, you will then just need to override the function Function EquipSuit(Actor akActor,String asEventName) with your own device suits

**Comment by kurotatsu77** [comment]
That esp has dependency to something called Cumshot - DD patch.esp.
What's the mod which contains it? I can't open the esp in xEdit while not all dependencies resolved...

**Comment by kurotatsu77** [comment, marked as answer]
OK, it was quite useful if it's only the script source which I managed to look through.
What I can say? I'm looking forward to what you're doing, it looks fun!
Also loving the auto-filling of items properties, really neat!

Now, what I suggest to write in documentation:
"To add custom Abadon suit you need to create control quest in which you have to apply 2 scripts - UD_PatchInit without any properties and your quest which extends UD_PatchInit and overrides function EquipSuit(target). This script should has following properties:
- UD_PatchName, contains String with patch name
- UD_CustomAbadonSuitEvent_1 ... UD_CustomAbadonSuitEvent_5, String, contains names of events, corresponding to suits
- UD_CustomAbadonSuitName_1 ... UD_CustomAbadonSuitName_5, String, contains names of the suits, up to 5 suits supported. If you need to add more than 5 suits - several quests has to be created."

And maybe attach small example - script and very simple esp in the archive.  This should  answer all questions.

**Comment by IHateMyKite** [comment]
Sorry about that, I forgot about the patch. I'm planning to release the patch once the DD beta is over. Here is the esp without the dependency on the patch. [UnforgivingDevices - Latex Witch Patch.zip](https://github.com/IHateMyKite/UnforgivingDevices/files/10158340/UnforgivingDevices.-.Latex.Witch.Patch.zip). But yeah, you understand it right. I will make separate tutorial about this feature once it will be part of the release.
